sql/sql_explain.cc

```diff
@@ -256,44 +259,75 @@ int Explain_query::print_explain_json(select_result_sink *output,
 #endif
 
   writer.start_object();
-  if (is_analyze && query_time_in_progress_ms > 0)
-    writer.add_member("r_query_time_in_progress_ms").
-           add_ull(query_time_in_progress_ms);
-
+  
   /*
     If we are printing ANALYZE FORMAT=JSON output, take into account that
     query's temporary tables have already been freed. See sql_explain.h,
     sql_explain.h:ExplainDataStructureLifetime for details.
   */
   if (is_analyze)
-    is_show_cmd= true;
+  {
+    if (query_time_in_progress_ms > 0){
+      writer.add_member("r_query_time_in_progress_ms").
+            add_ull(query_time_in_progress_ms);
+    }
+
+    print_query_optimization_json(&writer);
+    is_show_cmd = true;
+  }
+
+  bool were_query_plans_found = print_query_blocks_json(&writer, is_analyze, is_show_cmd);
 
+  writer.end_object();
+
+  if(were_query_plans_found){
+    send_explain_json_to_output(&writer, output);
+  }
+  
+  return 0;
+}
+
+void Explain_query::print_query_optimization_json(Json_writer *writer)
+{
+  if (query_optimization_total_time_tracker.has_timed_statistics())
+  {
+    // if more timers are added, move the query_optimization member to outside the if
+    writer->add_member("query_optimization").start_object();
+    writer->add_member("r_total_time_ms").
+            add_double(query_optimization_total_time_tracker.get_time_ms());
+    writer->end_object(); 
+  }
+}
+
+bool Explain_query::print_query_blocks_json(Json_writer *writer, const bool is_analyze, const bool is_show_cmd)
```

Comment:
Please keep the lines to be no longer than 80 characters.
(check the code around to see how multi-line code is formatted)